### PR TITLE
Feature/models

### PR DESCRIPTION
**Pull request by VeraLapidus**

### CODE REVIEW

File: auto_show/admin.py [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+class ActionAutoShowAdmin(admin.ModelAdmin):
+    """Класс для админки модели ActionAutoShow"""
+
+    list_display = ('name', 'amount_action', 'date_start', 'date_finish', 'auto_shows', 'is_active')
+    list_editable = ('is_active',)
+
+
+admin.site.register(AutoShow, AutoShowAdmin)
+admin.site.register(ActionAutoShow, ActionAutoShowAdmin)
+admin.site.register(DiscountAutoShow, DiscountAutoShowAdmin)
```
Comment (ksendzovSL) [inline auto_show/admin.py]: don't forget about blank line at the end of a file

File: auto_show/admin.py [hunk 2]
```diff
@@ -1,3 +1,28 @@
 from django.contrib import admin
 
-# Register your models here.
+from auto_show.models import AutoShow, ActionAutoShow, DiscountAutoShow
+
+
+class AutoShowAdmin(admin.ModelAdmin):
+    """Класс для админки модели AutoShow"""
```
Comment (ksendzovSL) [inline auto_show/admin.py]: doc strings must be in English

File: auto_show/filters.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from django_filters import rest_framework as filters
+
+from .models import AutoShow
```
Comment (ksendzovSL) [inline auto_show/filters.py]: don't use relative imports. use in all files:

from auto_show.models  import AutoShow

File: auto_show/models.py [hunk 4]
```diff
@@ -1,3 +1,85 @@
...
+
+    name = models.CharField(max_length=50, verbose_name='Имя')
+    amount_action = models.PositiveIntegerField(verbose_name='Скидка в %')
+    date_start = models.DateTimeField(verbose_name='Дата начала акции')
+    date_finish = models.DateTimeField(verbose_name='Дата окончания акции')
+    description = models.TextField(max_length=500, blank=True)
+
+    auto_shows = models.ForeignKey(AutoShow, on_delete=models.CASCADE, verbose_name='Автосалон')
+
+    is_active = models.BooleanField(default=True, verbose_name='Активен')
```
Comment (ksendzovSL) [inline auto_show/models.py]: in all models move same fields to Abstract Models

File: car/models.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    def __str__(self):
+        return f'{self.brand} {self.model} {self.year}'
+
+
+class CarInstance(models.Model):
+    """ Класс экземпляра автомобиля """
+
+    name = models.ForeignKey(Car, related_name='car_instances', on_delete=models.CASCADE, verbose_name="Автомобиль")
+    color = models.CharField(max_length=200, blank=True, null=True, verbose_name="Цвет")
+    CONDITION = [('wish_auto_show', 'желаемый для автосалона'), ('wish_customer', 'желаемый для покупателя'),
```
Comment (ksendzovSL) [inline car/models.py]: move to enums

File: car/views.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+class CarInstanceViewSet(mixins.CreateModelMixin,
+                 viewsets.ReadOnlyModelViewSet):
+    """ Вывод данных по экземплярам австомобилей """
+
+    queryset = CarInstance.objects.all()
+    serializer_class = CarInstanceSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    filter_backends = (rest_framework.DjangoFilterBackend, filters.SearchFilter, filters.OrderingFilter)
+    filterset_class = CarInstanceFilter  # http://127.0.0.1:8000/car/api/car_instance/?condition=wish_auto_show&max_price=130
```
Comment (ksendzovSL) [inline car/views.py:24]: delete comments in all files #http://127.0.0.1:8000/car..., as we have swagger

File: customer/admin.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+from customer.models import Customer
+
+class CustomerAdmin(admin.ModelAdmin):
+    """Класс для админки модели Customer"""
+
+    list_display = ('full_name', 'year_of_birth', 'balance', 'discount_auto_shows', 'is_active')
+    list_editable = ('is_active',)
+
+admin.site.register(Customer, CustomerAdmin)
```
Comment (ksendzovSL) [inline customer/admin.py]: blank line

File: customer/models.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    def full_name(self):
+        return "{} {}".format(self.last_name, self.first_name)
+
+    class Meta:
+        # ordering = ['last_name']
+        verbose_name = 'Покупатель'
+        verbose_name_plural = 'Покупатели'
+
+    def __str__(self):
+        return f'{self.last_name} {self.first_name}'
```
Comment (ksendzovSL) [inline customer/models.py]: blank line

File: my_showroom/urls.py [hunk 9]
```diff
@@ -13,9 +13,58 @@
...
+router = routers.DefaultRouter()
+router.register(r'deal', DealViewSet)
+router.register(r'car', CarViewSet)
+router.register(r'car_instance', CarInstanceViewSet)
+router.register(r'auto_show', AutoShowViewSet)
+router.register(r'customer', CustomerViewSet)
+router.register(r'producer', ProducerViewSet)
+
+# swagger
+schema_view = get_schema_view(
```
Comment (ksendzovSL) [inline my_showroom/urls.py]: make it under debug

if DEBUG:
    schema_view = ...

**Review by ksendzovSL — CHANGES_REQUESTED** [review]
whole comment:

it's a riddle
https://3dnews.ru/assets/external/illustrations/2020/05/01/1009921/01.jpg

File: auto_show/admin.py (hunk 1 above)
Comment (VeraLapidus) [reply auto_show/admin.py]: +

File: auto_show/admin.py (hunk 2 above)
Comment (VeraLapidus) [reply auto_show/admin.py]: ++

File: auto_show/filters.py (hunk 3 above)
Comment (VeraLapidus) [reply auto_show/filters.py]: ++

File: auto_show/models.py (hunk 4 above)
Comment (VeraLapidus) [reply auto_show/models.py]: ++

File: car/views.py (hunk 6 above)
Comment (VeraLapidus) [reply car/views.py:24]: ++

File: customer/admin.py (hunk 7 above)
Comment (VeraLapidus) [reply customer/admin.py]: ++

File: customer/models.py (hunk 8 above)
Comment (VeraLapidus) [reply customer/models.py]: ++

File: car/models.py (hunk 5 above)
Comment (VeraLapidus) [reply car/models.py]: ++

**Pull request merged**